src/main/steps/common/components/nationality.ts

```diff
@@ -0,0 +1,174 @@
+  [FieldPrefix.APPLICANT2]: APPLICANT_2_NATIONALITY,
+  [FieldPrefix.CHILDREN]: CHILDREN_NATIONALITY,
+};
+
+export const nationalityFields = (userCase: Partial<Case>, fieldPrefix: FieldPrefix): FormFields => ({
+  [`${fieldPrefix}Nationality`]: {
+    type: 'checkboxes',
+    label: l => l.label,
+    labelSize: 'l',
+    hint: l => l.hint,
+    // validator: atLeastOneFieldIsChecked,
+    validator: value =>
+      [FieldPrefix.APPLICANT1, FieldPrefix.APPLICANT1].includes(fieldPrefix)
+        ? atLeastOneFieldIsChecked(value)
+        : atLeastOneFieldIsChecked(value) || notSureViolation(value),
+    values: [
+      {
+        name: `${fieldPrefix}Nationality`,
+        label: l => l.british,
+        value: 'British',
+        hint: l => l.britishSubtext,
+      },
+      {
+        name: `${fieldPrefix}Nationality`,
+        label: l => l.irish,
+        value: 'Irish',
+      },
+      {
+        name: `${fieldPrefix}Nationality`,
+        label: l => l.differentCountry,
+        value: 'Other',
+        subFields: {
+          ...(userCase[`${fieldPrefix}AdditionalNationalities`]?.length
+            ? {
+                [`${fieldPrefix}AdditionalNationalities`]: {
+                  type: 'summarylist',
+                  values: [],
+                  rows: mapSummaryListContent(
+                    userCase[`${fieldPrefix}AdditionalNationalities`]!,
+                    ['Remove'],
+                    urls[fieldPrefix]
+                  ),
+                },
+              }
+            : {}),
+          ...(userCase[`${fieldPrefix}AdditionalNationalities`]?.length
+            ? {
+                addAnotherNationalityDetails: {
+                  type: 'details',
+                  label: l => l.another,
+                  subFields: {
+                    addAnotherNationality: {
+                      type: 'input',
+                      classes: 'govuk-!-width-two-thirds',
+                      label: l => l.countryName,
+                      labelSize: null,
+                    },
+                    addButton: {
+                      type: 'button',
+                      label: l => l.add,
+                      classes: 'govuk-button--secondary',
+                      value: 'addButton',
+                    },
+                  },
+                },
+              }
+            : {
+                addAnotherNationality: {
+                  type: 'input',
+                  classes: 'govuk-!-width-two-thirds',
+                  label: l => l.countryName,
+                  labelSize: null,
+                  validator: isFieldFilledIn,
+                },
+                addButton: {
+                  type: 'button',
+                  label: l => l.add,
+                  classes: 'govuk-button--secondary',
+                  value: 'addButton',
+                },
+              }),
+        },
+      },
+      ...(![FieldPrefix.APPLICANT1, FieldPrefix.APPLICANT1].includes(fieldPrefix)
```

Comment:
Please can you explain this logic? It looks like a possible mistake

Reply:
Applicant1 and Applicant2 must be british residents hence notSure option is not applicable for them

Reply:
In that case, should the array contain FieldPrefix.APPLICANT2?

Reply:
🙉 indeed yes